### PR TITLE
#159561794 User signup feature.

### PR DESCRIPTION
**Pull request by fnshemerirwe**
#### What does this PR do?
1. Creates a table to store the user's records.
2. Adds an endpoint that can be used by the user to signup.
3. Adds tests for the signup functionality.

[Pivotal Tracker story](https://www.pivotaltracker.com/story/show/159561794)

### CODE REVIEW

File: diary/views.py [hunk 1]
```diff
@@ -1,3 +1,49 @@
...
+        }
+        return Response(result, status=status.HTTP_201_CREATED)
+
+
+class ActivateAccount(APIView):
+    def get(self, request, slug_field=None):
+        token = request.GET.get('token')
+        try:
+            user = User.objects.get(slug_field=slug_field)
+            # print(ACTIVATIONTOKEN.check_token(user, token))
```
Comment (PatrickCmd) [inline diary/views.py]: I think this print statement should be removed.

File: my_diary/settings.py [hunk 2]
```diff
@@ -126,3 +128,20 @@
 # https://docs.djangoproject.com/en/2.1/howto/static-files/
 
 STATIC_URL = '/static/'
+
```
Comment (PatrickCmd) [inline my_diary/settings.py:131]: You also need to set the STATIC_ROOT variable which is also needed in production

**Review by PatrickCmd — COMMENTED** [review]
Did you set ALLOWED_HOSTS in your settings file, because it is very important for production

File: diary/views.py (hunk 1 above)
Comment (fnshemerirwe) [reply diary/views.py]: Thank you.

File: my_diary/settings.py (hunk 2 above)
Comment (fnshemerirwe) [reply my_diary/settings.py:131]: Thank you,  set the STATIC_ROOT in the subsequent tasks.

**Comment by fnshemerirwe** [comment]
@PatrickCmd , not yet, but I will set them once I have deployed on heroku.

**Pull request merged**